mission_control/navigator_missions/nav_missions/circle_tower.py

```diff
@@ -7,55 +7,114 @@
 from sensor_msgs.msg import PointCloud
 import datetime
 
-@txros.util.cancellableInlineCallbacks
-def main(navigator):
-    res = navigator.fetch_result()
-
-    buoy_field = yield navigator.database_query("BuoyField")
-    buoy_field_point = navigator_tools.point_to_numpy(buoy_field.objects[0])
 
-    #yield navigator.move.set_position(buoy_field_point).go()
+BF_WIDTH = 60.0  # m
+BF_EST_COFIDENCE = 10.0  # How percisly can they place the waypoints? (m)
+TOTEM_SAFE_DIST = 7  # How close do we go to the totem
+ROT_SAFE_DIST = 3.5  # How close to rotate around it
 
-    circle_colors = ['blue', 'red']
-    color_map = {'blue': (255, 0, 0), 'red': (0, 0, 255)}
+@txros.util.cancellableInlineCallbacks
+def main(navigator, **kwargs):
+    # rgb color map to param vlues
+    color_map = {'BLUE': [0, 0, 1], 'RED': [1, 0, 0], 'YELLOW': [1, 1, 0], 'GREEN': [0, 1, 0]}
 
     explored_ids = []
     all_found = False
 
-    while not all_found:
-        target_totem, explored_ids = yield get_closest_buoy(navigator, explored_ids)
+    # Get colors of intrest and directions
+    c1 = navigator.mission_params['totem_color_1'].get()
+    d1 = navigator.mission_params['totem_direction_1'].get()
+    c2 = navigator.mission_params['totem_color_2'].get()
+    d2 = navigator.mission_params['totem_direction_2'].get()
+    
+    colors = [c1, c2]
```

Comment:
The rules are bad, but I think there will be 3 to circle, and maybe 4 if they feel like it.